src/app/credExplorer/pagerankTable/Aggregation.js

```diff
@@ -0,0 +1,135 @@
+// @flow
+
+import React from "react";
+import * as NullUtil from "../../../util/null";
+
+import type {NodeAddressT} from "../../../core/graph";
+import {ConnectionRowList} from "./Connection";
+
+import {aggregateFlat, type FlatAggregation} from "./aggregate";
+
+import {Badge, type SharedProps} from "./shared";
+import {TableRow} from "./TableRow";
+
+type AggregationRowListProps = {|
+  +depth: number,
+  +node: NodeAddressT,
+  +sharedProps: SharedProps,
+|};
+
+export class AggregationRowList extends React.PureComponent<
+  AggregationRowListProps
+> {
+  render() {
+    const {depth, node, sharedProps} = this.props;
+    const {pnd, adapters} = sharedProps;
+    const {scoredConnections} = NullUtil.get(pnd.get(node));
+    const aggregations = aggregateFlat(
+      scoredConnections,
+      adapters.static().nodeTypes(),
+      adapters.static().edgeTypes()
+    );
+    function aggregationKey(aggregation: FlatAggregation) {
```

Comment:
I don’t understand the implications of this function. Why can’t its
implementation just be

```js
    function aggregationKey(aggregation: FlatAggregation) {
      const result = {
        nodePrefix: aggregation.nodeType.prefix,
        connectionType: aggregation.connectionType.type,
      };
      return stringify(result);
    }
```

? This passes Flow and tests, and seems to work under `yarn start`. Are
additional considerations that should be taken into account?

Reply:
This would not be unique. Note that the connection type is just a string in `"IN_EDGE" | "OUT_EDGE" | "SYNTHETIC_LOOP"`. Because the edge prefix is not included, we can get a conflict when there are multiple edge types pointing to the same node type.

I've factored `aggregationKey` into an exported function from `aggregate.js`, and tested it. Using your proposed implementation would properly fail the test. (This happens in a separate commit; I'm planning to rebase and merge them)

Reply:
Great. Thanks for the explanation.